keps/sig-cluster-lifecycle/0015-kubeadm-join-master.md

```diff
@@ -0,0 +1,431 @@
+
+Among them, the approach best suited for a dynamic bootstrap workflow requires the
+user to set the `--apiserver-advertise-address` of each master, including the bootstrap master
+itself, equal to the `controlplaneAddress` endpoint provided during kubeadm init
+(the IP/DNS of the external load balancer).
+
+By using the same advertise address for all the IP masters, `kubeadm init` can create
+a unique API server serving certificate that could be shared across many masters nodes;
+no changes will be required to this certificate when adding/removing master nodes.
+
+Please note that:
+
+- if the user is not planning to distribute the apiserver serving certificate among masters,
+  kubeadm will generate a new apiserver serving certificate “almost equal” to the certificate
+  created on the bootstrap master (it differs only for the domain name of the joining master)
+
+#### Static workflow (advertise-address != `controlplaneAddress`)
+
+In case of a static bootstrap workflow the final layout of the controlplane - the number, the
+name and the IP of master nodes - is know in advance.
+
+Given such information, the user can choose a different approach where each master has a
+specific apiserver advertise address different from the `controlplaneAddress`.
+
+Please note that:
+
+- if the user is not planning to distribute the apiserver certificate among masters, kubeadm
+  will generate a new apiserver serving certificate with the required SANS
+- if the user is planning to distribute the apiserver certificate among masters, he/she/the
+  operator is required to provide during `kubeadm init` the list of masters/the list of IP
+  addresses for all the masters as alternative names for the API servers certificate, thus
+  allowing the proper functioning of all the API server instances that will join
+
+#### Strategies for deploying control plane components
+
+As of today kubeadm supports two solutions for deploying control plane components:
+
+1. Control plane deployed as static pods (current kubeadm default)
+2. Self-hosted control plane in case of `--feature-gates=SelfHosting=true`
+
+The proposed solution for case 1. "Control plane deployed as static pods", assumes
+that the `kubeadm join --master` flow will take care of creating required kubeconfig
+files and required static pod manifests.
+
+Self-hosted control plane are not supported in this release of the proposal
+(`kubeadm join --master` will fail immediately), but nothing in this proposal should
+prevent to address this in subsequent phases.
+
+#### Strategies for distributing cluster certificates
+
+As of today kubeadm supports two solutions for storing cluster certificates:
+
+1. Cluster certificates stored on file system in case of:
+   - Control plane deployed as static pods (current kubeadm default)
+   - Self-hosted control plane in case of `--feature-gates=SelfHosting=true,StoreCertsInSecrets=false`
+2. Cluster certificates stored in secrets in case of:
```

Comment:
We actually don't just alpha code lying around. Please remove this consideration